src/features/hotspots/setup/HotspotSetupAddTxnScreen.tsx

```diff
@@ -0,0 +1,168 @@
+import { useNavigation } from '@react-navigation/native'
+import React, { useEffect, useMemo, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useSelector } from 'react-redux'
+import BackScreen from '../../../components/BackScreen'
+import Box from '../../../components/Box'
+import Button from '../../../components/Button'
+import Text from '../../../components/Text'
+import { useConnectedHotspotContext } from '../../../providers/ConnectedHotspotProvider'
+import { RootState } from '../../../store/rootReducer'
+import { currencyFormat } from '../../../utils/currency'
+import { calculatePaymentTxnFee } from '../../../utils/transactions'
+import usePermissionManager from '../../../utils/usePermissionManager'
+import { HotspotSetupNavigationProp } from './hotspotSetupTypes'
+
+const HotspotSetupAddTxnScreen = () => {
+  const navigation = useNavigation<HotspotSetupNavigationProp>()
+  const { t } = useTranslation()
+  const [fee, setFee] = useState(0)
+  const [submitting, setSubmitting] = useState(false)
+  const { updateHotspotStatus, addGatewayTxn } = useConnectedHotspotContext()
+
+  const {
+    account: { account },
+    connectedHotspot: { mac, freeAddHotspot, status: hotspotStatus },
+  } = useSelector((state: RootState) => state)
+  const { hasLocationPermission } = usePermissionManager()
+
+  const navNext = async () => {
+    if (hotspotStatus === 'global') {
+      navigation.popToTop()
+      return
+    }
+
+    if (hotspotStatus === 'owned' || hotspotStatus === 'new') {
+      const hasLocPerm = await hasLocationPermission()
+      if (hasLocPerm) navigation.push('HotspotLocationFeeScreen')
+      else navigation.push('EnableLocationScreen')
+    }
+  }
+
+  const act = async () => {
+    if (hotspotStatus === 'new') {
+      setSubmitting(true)
+
+      const txn = await addGatewayTxn()
+      setSubmitting(false)
+      if (txn) {
+        navNext()
+      }
+    } else {
+      navNext()
+    }
+  }
+
+  useEffect(() => {
+    // TODO: Verify this is correct
+    if (freeAddHotspot) {
+      setFee(0)
+      return
+    }
+
+    const calc = async () => {
+      const nonce = account?.nonce ? account.nonce + 1 : 0
+      // TODO:                                            ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
+      const val = await calculatePaymentTxnFee(0, nonce, 'what should this be?')
```

Comment:
I think we'd want to be calculating the fee for an `add_gateway_v1` txn rather than payment. But since txn fee is based on byte size of the serialized txn, I think I just hardcoded the fee and staking amount for add and assert, because it'll always be the same. Payment txn fee can change depending on how large the `amount` integer is and how many recipients you add (although our app only supports one recipient right now)

Reply:
I just looked back at the old app. The fee calculation looks like this
```
  calculateFee = () => {
    const {
      data: {
        account: { fee },
      },
      bluetooth: { connectedHotspotDetails },
    } = this.props
    if (connectedHotspotDetails.free_add_hotspot) return 0
    return fee
  }
  ```
  
But, [fee in the account data is hardcoded to 0](https://github.com/helium/mobile-wallet/blob/master/src/data/AppData.js#L311)

Can I assume free at this point?

Reply:
yeah I mean to the user, the `add_gateway` fee will always be 0, for now. Either they're allowed to proceed or not